File: 782-jewels-and-stones/jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        final var jewelsSet = new HashSet<Character>();
        
        for (int i = 0; i < jewels.length(); i++) {
            jewelsSet.add(jewels.charAt(i));
        }
        
        int count = 0;
        
        for (int i = 0; i < stones.length(); i++) {
            if (jewelsSet.contains(stones.charAt(i))) {
                count++;
            }
        }
        
        return count;
    }
}

File: 782-jewels-and-stones/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewels_set = { ch for ch in jewels }
        count = 0
        for ch in stones:
            if ch in jewels_set:
                count += 1
        
        return count
                